### PR TITLE
update `GutenbergSettings` to interact with CoreData safely

### PR DESCRIPTION
**Pull request by crazytonyli**
There appears to be no requirement about the thread from which `GutenbergSettings` should be called. But, some methods uses the `mainContext` directly. This PR removes the `mainContext` reference, by using `CoreDataStack.performQuery` for reads and `CoreDataStack.performAndSave` for writes.

BTW, the `EditorSettingsService` will also be refactored away from using `NSManagedObjectContext` directly, which will be in a separate PR.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None.

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/Utility/Editor/GutenbergSettings.swift [hunk 1]
```diff
@@ -81,14 +79,14 @@ class GutenbergSettings {
     }
 
     private func setGutenbergEnabledForAllSites() {
-        let allBlogs = (try? BlogQuery().blogs(in: context)) ?? []
+        let allBlogs = Environment.current.contextManager.performQuery({ (try? BlogQuery().blogs(in: $0)) ?? [] })
```
Comment (mokagio) [inline WordPress/Classes/Utility/Editor/GutenbergSettings.swift]: Did you consider defining a computed `var` for the context manager, to streamline the coding a bit?

```swift
var contextManager: CoreDataStack { Environment.current.contextManager }

// ...

let allBlogs = contextManager.performQuery...

**Review by mokagio — APPROVED** [review]
This looks good to me, approving

Before merging, I'd like to ping @fluiddot @dcalhoun @SiobhyB as folks knowledgeable about Gutenberg to give a confidence check on the threading requirement, or top ping who could help. Thanks!

**Comment by mokagio** [comment]
I could be wrong, but I don't think this is a must have for 21.7. As such, I'll bump the milestone to 21.8 and wait for feedback by the Gutenberg folks.

I'll take care of cherry-picking in 21.7 if my assessment turns out incorrect.

File: WordPress/Classes/Utility/Editor/GutenbergSettings.swift (hunk 1 above)
Comment (crazytonyli) [reply WordPress/Classes/Utility/Editor/GutenbergSettings.swift]: That's a good idea! Updated in https://github.com/wordpress-mobile/WordPress-iOS/pull/20044/commits/7bd6f006b25480fe50eeca2b09bc40814e360477.

**Review by dcalhoun — COMMENTED** [review]
While this PR is already merged, I wanted to leave a note for posterity. I am not intimately familiar with the modified code, but I tested the build attached to this PR and did not observe any issues with the editor. 🎉 

Additionally, the modifications (1) do not appear to conflict with the [React Native modules guidance](https://reactnative.dev/docs/0.69/native-modules-ios#threading) and (2) I do not believe this code directly relates to a bridging module.

**Pull request merged**